tests/qchem/test_tapering.py

```diff
@@ -629,3 +631,265 @@ def test_taper_obs(symbols, geometry, charge):
+    paulixops = qml.paulix_ops(generators, len(hamiltonian.wires))
+    paulix_sector = optimal_sector(hamiltonian, generators, mol.n_electrons)
+    wire_order = hamiltonian.wires
+
+    with pytest.raises(Exception, match=message_match):
+        taper_operation(operation, generators, paulixops, paulix_sector, wire_order, gen_op)
+
+
+@pytest.mark.parametrize(
+    ("operation", "gen_op"),
+    [
+        (qml.PauliX(1), qml.Hamiltonian((-np.pi / 2,), [qml.PauliX(wires=[1])])),
+        (qml.PauliZ(3), qml.Hamiltonian((-np.pi / 2,), [qml.PauliZ(wires=[3])])),
+        (
+            qml.OrbitalRotation(1, wires=[0, 1, 2, 3]),
+            qml.Hamiltonian(
+                (0.25, -0.25, 0.25, -0.25),
+                [
+                    qml.PauliX(wires=[0]) @ qml.PauliY(wires=[2]),
+                    qml.PauliY(wires=[0]) @ qml.PauliX(wires=[2]),
+                    qml.PauliX(wires=[1]) @ qml.PauliY(wires=[3]),
+                    qml.PauliY(wires=[1]) @ qml.PauliX(wires=[3]),
+                ],
+            ),
+        ),
+    ],
+)
+def test_consistent_taper_ops(operation, gen_op):
+    r"""Test that operations are tapered consistently when their generators are provided manually and when they are constructed internally"""
+
+    # HeH+ tapering data for [[0.0, 0.0, 0.0], [0.0, 0.0, 1.4588684632]]
+    generators = [
+        qml.Hamiltonian([1.0], [qml.PauliZ(0) @ qml.PauliZ(2)]),
+        qml.Hamiltonian([1.0], [qml.PauliZ(1) @ qml.PauliZ(3)]),
+    ]
+    paulixops = [qml.PauliX(wires=[2]), qml.PauliX(wires=[3])]
+    paulix_sector = (-1, -1)
+    wire_order = range(4)
+
+    taper_op1 = taper_operation(operation, generators, paulixops, paulix_sector, wire_order, None)
+    taper_op2 = taper_operation(operation, generators, paulixops, paulix_sector, wire_order, gen_op)
+    assert np.all([qml.equal(op1, op2) for op1, op2 in zip(taper_op1, taper_op2)])
+
+    tape1, tape2 = qml.tape.QuantumTape(), qml.tape.QuantumTape()
+    with tape1:
+        taper_operation(operation, generators, paulixops, paulix_sector, wire_order, None)
+    with tape2:
+        taper_operation(operation, generators, paulixops, paulix_sector, wire_order, gen_op)
+
+    taper_circuit1 = [x for x in tape1.circuit if x.label() != "I"]
+    taper_circuit2 = [x for x in tape2.circuit if x.label() != "I"]
+
+    assert len(taper_op1) == len(taper_circuit1) and len(taper_op2) == len(taper_circuit2)
+    assert np.all([qml.equal(op1, op2) for op1, op2 in zip(taper_circuit1, taper_op1)])
+    assert np.all([qml.equal(op1, op2) for op1, op2 in zip(taper_circuit2, taper_op2)])
+
+
+@pytest.mark.parametrize(
+    ("symbols", "geometry", "charge", "fci_energy"),
+    [
+        (
+            ["He", "H"],
+            np.array(
+                [[0.0, 0.0, 0.0], [0.0, 0.0, 1.4588684632]],
+                requires_grad=True,
+            ),
+            1,
+            -2.85102402,
+        ),
+    ],
+)
+def test_tapered_vqe(symbols, geometry, charge, fci_energy):
```

Comment:
I'm not sure this integration test makes sense to include here. It's good to make these checks during development, but it's problematic to include such expensive tests in the code. Please remove this